### ISSUE TITLE
Custom columns settings should not be reset to defaults when clicking Build Table buttons

### ISSUE THREAD

**Issue by martinholmer** [OpenSourcePolicyCenter/PolicyBrain#682]
The TaxBrain static results page has been much improved with the enhancements made by @GoFroggyRun.  So, many thanks to Sean for all his work.

But I've been saving until the end the aspect of the Build-Table/Select-Columns section (at the bottom of the results page below the annual revenue estimates) that is the most annoying.  Part of the reason it is very annoying is that it is time consuming to work around.  It makes what would otherwise be an easy-to-use set of buttons and ckeck boxes, into an awkward interface.

There are two different kinds of tables: distribution tables and difference tables, each of which has its own set of possible columns.  There is a default setting for the columns used in each type of table.  The distribution tables have 12 of the 19 possible columns selected as the default, while the difference tables have 8 of the 8 possible columns selected as the default.

The problem arises when a TaxBrain user customizes the columns (s)he want included in the table.
A typical analysis scenario is that the user then wants several tables all of that type but with different choices for the taxes to include, the income concept to use (Expanded Income vs AGI), and income categories (bins vs deciles).  Every time the user clicks on one of those buttons, the customized column selection goes away and the default column selection is reimposed on the user.  

This is very poor etiquette on the part of TaxBrain.  In fact, it is approaching abusive behavior on the part of TaxBrain.

Why can't TaxBrain respect the column choices a user makes when the user wants to see a different table of the same type?

@GoFroggyRun, can you give this enhancement a high priority for the next TaxBrain release?

@MattHJensen @hdoupe @brittainhard

**Comment by GoFroggyRun** [comment]
Thanks for pointing out the problem with the "Select Columns" options @martinholmer. 

I agree that "This is very poor etiquette on the part of TaxBrain." However, given how tables and data are currently organized in TaxBrain, I found it not trivial to eliminate such re-imposed default. Each button option, namely "Distribution" vs "Difference", "Current Law" vs "Reform", and "Income Bins" vs "Income Deciles", is associated with different data sources, where each data source is equipped with its own "Select Columns" table. Whenever columns were selected/unselected for certain data source, such customization will be remembered and re-imposed even after different tables were rendered (and of course user rendered back to that customized table). The "re-imposing default" behavioral is rooted in the fact that we are having multiple defaults (or each data source has its own default).

I tried to "unify" the defaults, but didn't make much headway. On the other hand, @MattHJensen, @hdoupe and I have some premeditated "result page overhaul" that intends to simplify TaxBrain output page. A couple interfaces (like the "Select Columns") will be eliminated, and visualizations will be filled instead. I believe the annoyance caused by issue #682 will be naturally resolved once the renovation is finished.
